src/Featurizers/Components/FrequencyEstimator.h

```diff
@@ -0,0 +1,169 @@
+
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         FrequencyAnnotation
+///  \brief         This is an annotation class which holds the frequency
+///                 of TimeSeries
+///
+class FrequencyAnnotation{
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyType                     = std::chrono::system_clock::duration;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyType                           Value;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyAnnotation(FrequencyType value);
+    ~FrequencyAnnotation(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(FrequencyAnnotation);
+};
+
+namespace Details {
+    
+/////////////////////////////////////////////////////////////////////////
+///  \class         FrequencyTrainingOnlyPolicy
+///  \brief         `FrequencyEstimator` implementation details.
+///
+class FrequencyTrainingOnlyPolicy {
+public:
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                     = std::chrono::system_clock::time_point;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = FrequencyEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    FrequencyAnnotation complete_training(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyType                     = std::chrono::system_clock::duration;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyType _frequency;
+    InputType _lastObserved;
+
+    bool _hasSeenValues;
+};
+}

```

Comment:
Namespace comment here.

Reply:
good catch!

Reply:
This still needs to be addressed.

---
In reply to: [388728448](https://github.com/microsoft/FeaturizersLibrary/pull/160#discussion_r388728448) [](ancestors = 388728448)